### ISSUE TITLE
faker.pydict() sometimes fail to generate correct dates

### ISSUE THREAD

**Issue by thedrow** [joke2k/faker#282]
```
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/base.py:571: in create_batch
    return [cls.create(**kwargs) for _ in range(size)]
../../../../common/testing/factories.py:216: in create
    return super(SampleFactory, cls).create(**kwargs)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/base.py:558: in create
    attrs = cls.attributes(create=True, extra=kwargs)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/base.py:400: in attributes
    force_sequence=force_sequence,
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/containers.py:228: in build
    return stub.__fill__()
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/containers.py:83: in __fill__
    res[attr] = getattr(self, attr)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/containers.py:105: in __getattr__
    val = val.evaluate(self, self.__containers)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/containers.py:160: in evaluate
    containers=containers,
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/factory/fuzzy.py:66: in evaluate
    return self.fuzz()
../../../../common/testing/fuzzy.py:164: in fuzz
    return self.faker.pydict()
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/faker/providers/python/__init__.py:86: in pydict
    self._pyiterable(nb_elements, False, *value_types)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/faker/providers/python/__init__.py:74: in _pyiterable
    yield self._random_type(value_types)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/faker/providers/python/__init__.py:59: in _random_type
    return self.generator.format(value_type)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/faker/generator.py:61: in format
    return self.get_formatter(formatter)(*args, **kwargs)
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/faker/providers/date_time/__init__.py:251: in date_time
    return datetime.fromtimestamp(cls.unix_time())
../../../../.tox/py27-malware-functional/lib/python2.7/site-packages/faker/providers/date_time/__init__.py:234: in unix_time
    return random.randint(0, int(time()))
/home/omer/.pyenv/versions/2.7.10/lib/python2.7/random.py:242: in randint
    return self.randrange(a, b+1)
self = <random.Random object at 0x1a77920>, start = 0, stop = -10325, step = 1
_int = <type 'int'>, _maxwidth = 9007199254740992L

    def randrange(self, start, stop=None, step=1, _int=int, _maxwidth=1L<<BPF):
        """Choose a random item from range(start, stop[, step]).
    
            This fixes the problem with randint() which includes the
            endpoint; in Python this is usually not what you want.
    
            """
    
        # This code is a bit messy to make it fast for the
        # common case while still doing adequate error checking.
        istart = _int(start)
        if istart != start:
            raise ValueError, "non-integer arg 1 for randrange()"
        if stop is None:
            if istart > 0:
                if istart >= _maxwidth:
                    return self._randbelow(istart)
                return _int(self.random() * istart)
            raise ValueError, "empty range for randrange()"
    
        # stop argument supplied.
        istop = _int(stop)
        if istop != stop:
            raise ValueError, "non-integer stop for randrange()"
        width = istop - istart
        if step == 1 and width > 0:
            # Note that
            #     int(istart + self.random()*width)
            # instead would be incorrect.  For example, consider istart
            # = -2 and istop = 0.  Then the guts would be in
            # -2.0 to 0.0 exclusive on both ends (ignoring that random()
            # might return 0.0), and because int() truncates toward 0, the
            # final result would be -1 or 0 (instead of -2 or -1).
            #     istart + int(self.random()*width)
            # would also be incorrect, for a subtler reason:  the RHS
            # can return a long, and then randrange() would also return
            # a long, but we're supposed to return an int (for backward
            # compatibility).
    
            if width >= _maxwidth:
                return _int(istart + self._randbelow(width))
            return _int(istart + _int(self.random()*width))
        if step == 1:
>           raise ValueError, "empty range for randrange() (%d,%d, %d)" % (istart, istop, width)
E           ValueError: empty range for randrange() (0,-10325, -10325)
```
The stop parameter should always be higher than the start parameter.

**Comment by fcurella** [comment]
I'm a little confused on how this could happen.

From your stracktrace, it looks like `randrange()` has been called with `(0,-10325, -10325)` from `randint(0, int(time()))`.

I don't understand how the value of `int(time())` could become `-10325` (or any value < 0). What OS are you using?

**Comment by thedrow** [comment]
It's Linux. Ubuntu 15.04 and Ubuntu 12.04.
But now that I think about it, I see it way more often on Travis than on my laptop.

**Comment by fcurella** [comment]
fixed in #307

**Comment by thedrow** [comment]
Thanks!
